Compilation issue

Hi,

I am very new with Sparkoscope and I am trying to get it working.
After compiling Spark using command:
./build/mvn -Pyarn -Phadoop-2.7 -Dhadoop.version=2.7.3 -DskipTests clean package

I could not start Spark-shell properly. Running bin/spark-shell will raise an exception "java.lang.VerifyError: class io.netty.channel.nio.NioEventLoop overrides final method pendingTasks"

Did anyone happen to have the same issue? Thanks.

====================================
Sparkoscope 2.1: spark-shell raises an exception
====================================
sparkoscope$ bin/spark-shell
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
17/08/15 14:03:04 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
17/08/15 14:03:04 WARN Utils: Your hostname, liam resolves to a loopback address: 127.0.1.1; using 10.129.2.113 instead (on interface eth0)
17/08/15 14:03:04 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
java.lang.VerifyError: class io.netty.channel.nio.NioEventLoop overrides final method pendingTasks.()I
  at java.lang.ClassLoader.defineClass1(Native Method)
  at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
  at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
  at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
  at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
  at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
  at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
  at java.security.AccessController.doPrivileged(Native Method)
  at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
  at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
  at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
  at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
  at io.netty.channel.nio.NioEventLoopGroup.newChild(NioEventLoopGroup.java:102)
  at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:64)
  at io.netty.channel.MultithreadEventLoopGroup.<init>(MultithreadEventLoopGroup.java:49)
  at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:70)
  at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:65)
  at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:56)
  at org.apache.spark.network.util.NettyUtils.createEventLoop(NettyUtils.java:51)
  at org.apache.spark.network.client.TransportClientFactory.<init>(TransportClientFactory.java:104)
  at org.apache.spark.network.TransportContext.createClientFactory(TransportContext.java:89)
  at org.apache.spark.rpc.netty.NettyRpcEnv.<init>(NettyRpcEnv.scala:70)
  at org.apache.spark.rpc.netty.NettyRpcEnvFactory.create(NettyRpcEnv.scala:442)
  at org.apache.spark.rpc.RpcEnv$.create(RpcEnv.scala:56)
  at org.apache.spark.SparkEnv$.create(SparkEnv.scala:245)
  at org.apache.spark.SparkEnv$.createDriverEnv(SparkEnv.scala:174)
  at org.apache.spark.SparkContext.createSparkEnv(SparkContext.scala:257)
  at org.apache.spark.SparkContext.<init>(SparkContext.scala:432)
  at org.apache.spark.SparkContext$.getOrCreate(SparkContext.scala:2313)
  at org.apache.spark.sql.SparkSession$Builder$$anonfun$6.apply(SparkSession.scala:868)
  at org.apache.spark.sql.SparkSession$Builder$$anonfun$6.apply(SparkSession.scala:860)
  at scala.Option.getOrElse(Option.scala:121)
  at org.apache.spark.sql.SparkSession$Builder.getOrCreate(SparkSession.scala:860)
  at org.apache.spark.repl.Main$.createSparkSession(Main.scala:101)
  ... 47 elided
<console>:14: error: not found: value spark
       import spark.implicits._
              ^
<console>:14: error: not found: value spark
       import spark.sql
              ^
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /___/ .__/\_,_/_/ /_/\_\   version 2.1.0
      /_/
         
Using Scala version 2.11.8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_131)
Type in expressions to have them evaluated.
Type :help for more information.

scala> 

====================================
Spark 2.2 downloaded from Spark website and complied: spark-shell works fine
====================================
/spark-2.2.0$ bin/spark-shell
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
17/08/15 14:00:47 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
17/08/15 14:00:47 WARN Utils: Your hostname, liam resolves to a loopback address: 127.0.1.1; using 10.129.2.113 instead (on interface eth0)
17/08/15 14:00:47 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
Spark context Web UI available at http://10.129.2.113:4040
Spark context available as 'sc' (master = local[*], app id = local-1502830848390).
Spark session available as 'spark'.
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /___/ .__/\_,_/_/ /_/\_\   version 2.2.0
      /_/
         
Using Scala version 2.11.8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_131)
Type in expressions to have them evaluated.
Type :help for more information.

scala>

Update:
I tried the pre-compiled 1.6.2 that @YiannisGkoufas uploaded and it works as well.
https://github.com/ibm-research-ireland/sparkoscope/releases/download/v1.6.2/spark-1.6.2-bin-sparkoscope.tgz

However, trying to compile sparkoscope 1.6.2 will give me an error at **Spark Project Parent POM**:
"[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce (enforce-versions) on project spark-parent_2.10: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed."

--- a/core/pom.xml
+++ b/core/pom.xml
@@ -356,6 +356,32 @@
       <groupId>io.moquette</groupId>
       <artifactId>moquette-broker</artifactId>
       <version>0.8.1</version>
+      <exclusions>
+        <exclusion>
+          <groupId>io.netty</groupId>
+          <artifactId>netty-common</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>io.netty</groupId>
+          <artifactId>netty-buffer</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>io.netty</groupId>
+          <artifactId>netty-transport</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>io.netty</groupId>
+          <artifactId>netty-handler</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>io.netty</groupId>
+          <artifactId>netty-codec</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>io.netty</groupId>
+          <artifactId>netty-codec-http</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>org.eclipse.paho</groupId>

